### ISSUE TITLE
Running into Unauthorized error while creating a new SMART service

### ISSUE THREAD

**Issue by JorisHeadease** [Koppeltaal/KT2-Docker-Compose#2]
Rather easy fix:

```
koppeltaal_2_docker_compose-domain-admin-1  | 2023-06-05 16:42:23.155 ERROR 8 --- [nio-8080-exec-5] n.k.s.b.s.s.event.AutomatedAuditEvents   : Failed to validate resource [org.hl7.fhir.r4.model.Device] and id [null] with the following message: {
koppeltaal_2_docker_compose-domain-admin-1  |   "resourceType": "OperationOutcome",
koppeltaal_2_docker_compose-domain-admin-1  |   "text": {
koppeltaal_2_docker_compose-domain-admin-1  |     "status": "generated",
koppeltaal_2_docker_compose-domain-admin-1  |     "div": "<div xmlns=\"http://www.w3.org/1999/xhtml\"><h1>Operation Outcome</h1><table border=\"0\"><tr><td style=\"font-weight: bold;\">WARNING</td><td>[Device, Line 1, Col 2]</td><td><pre>Rule dom-6: 'A resource should have narrative for robust management' Failed</pre></td>\n\t\t\t</tr>\n\t\t\t<tr>\n\t\t\t\t<td style=\"font-weight: bold;\">ERROR</td>\n\t\t\t\t<td>[Device.meta.source, Line 1, Col 91]</td>\n\t\t\t\t<td><pre>UUIDs must be valid (and lowercase)</pre></td>\n\t\t\t</tr>\n\t\t\t<tr>\n\t\t\t\t<td style=\"font-weight: bold;\">ERROR</td>\n\t\t\t\t<td>[Device.meta.profile[0], Line 1, Col 2]</td>\n\t\t\t\t<td><pre>Profile reference 'http://koppeltaal.nl/fhir/StructureDefinition/KT2Device' has not been checked because it is unknown, and the validator is set to not fetch unknown profiles</pre></td>\n\t\t\t</tr>\n\t\t</table>\n\t</div>"
koppeltaal_2_docker_compose-domain-admin-1  |   },
koppeltaal_2_docker_compose-domain-admin-1  |   "issue": [ {
koppeltaal_2_docker_compose-domain-admin-1  |     "severity": "warning",
koppeltaal_2_docker_compose-domain-admin-1  |     "code": "processing",
koppeltaal_2_docker_compose-domain-admin-1  |     "details": {
koppeltaal_2_docker_compose-domain-admin-1  |       "coding": [ {
koppeltaal_2_docker_compose-domain-admin-1  |         "system": "http://hl7.org/fhir/java-core-messageId",
koppeltaal_2_docker_compose-domain-admin-1  |         "code": "Rule dom-6: 'A resource should have narrative for robust management' Failed"
koppeltaal_2_docker_compose-domain-admin-1  |       } ]
koppeltaal_2_docker_compose-domain-admin-1  |     },
koppeltaal_2_docker_compose-domain-admin-1  |     "diagnostics": "Rule dom-6: 'A resource should have narrative for robust management' Failed",
koppeltaal_2_docker_compose-domain-admin-1  |     "location": [ "Device", "Line 1, Col 2" ]
koppeltaal_2_docker_compose-domain-admin-1  |   }, {
koppeltaal_2_docker_compose-domain-admin-1  |     "severity": "error",
koppeltaal_2_docker_compose-domain-admin-1  |     "code": "processing",
koppeltaal_2_docker_compose-domain-admin-1  |     "details": {
koppeltaal_2_docker_compose-domain-admin-1  |       "coding": [ {
koppeltaal_2_docker_compose-domain-admin-1  |         "system": "http://hl7.org/fhir/java-core-messageId",
koppeltaal_2_docker_compose-domain-admin-1  |         "code": "Type_Specific_Checks_DT_UUID_Valid"
koppeltaal_2_docker_compose-domain-admin-1  |       } ]
koppeltaal_2_docker_compose-domain-admin-1  |     },
koppeltaal_2_docker_compose-domain-admin-1  |     "diagnostics": "UUIDs must be valid (and lowercase)",
koppeltaal_2_docker_compose-domain-admin-1  |     "location": [ "Device.meta.source", "Line 1, Col 91" ]
koppeltaal_2_docker_compose-domain-admin-1  |   }, {
koppeltaal_2_docker_compose-domain-admin-1  |     "severity": "error",
koppeltaal_2_docker_compose-domain-admin-1  |     "code": "processing",
koppeltaal_2_docker_compose-domain-admin-1  |     "details": {
koppeltaal_2_docker_compose-domain-admin-1  |       "coding": [ {
koppeltaal_2_docker_compose-domain-admin-1  |         "system": "http://hl7.org/fhir/java-core-messageId",
koppeltaal_2_docker_compose-domain-admin-1  |         "code": "VALIDATION_VAL_PROFILE_UNKNOWN_NOT_POLICY"
koppeltaal_2_docker_compose-domain-admin-1  |       } ]
koppeltaal_2_docker_compose-domain-admin-1  |     },
koppeltaal_2_docker_compose-domain-admin-1  |     "diagnostics": "Profile reference 'http://koppeltaal.nl/fhir/StructureDefinition/KT2Device' has not been checked because it is unknown, and the validator is set to not fetch unknown profiles",
koppeltaal_2_docker_compose-domain-admin-1  |     "location": [ "Device.meta.profile[0]", "Line 1, Col 2" ]
koppeltaal_2_docker_compose-domain-admin-1  |   } ]
koppeltaal_2_docker_compose-domain-admin-1  | }

```

**Comment by JorisHeadease** [comment]
The `meta.source` seems easy, but the missing profile is interesting.

The loaded npm package should install the KT2 profiles, like https://simplifier.net/packages/koppeltaalv2.00/0.9.4/files/1966211

However, it seems just `nl-core` and the `zibs` are installed.

**Comment by JorisHeadease** [comment]
It actually seems to be a "problem" with the latest release. Looking at the previous versions, all resources seem to have 2 entries in the npm package: https://simplifier.net/packages/Koppeltaalv2.00/0.9.3/~files.

Looking at the latest release, there is only one: https://simplifier.net/packages/Koppeltaalv2.00/0.9.4/~files. While this seems more logical, it is the difference between loading or not loading the `StructureDefinitions` in the FHIR store.

**Comment by JorisHeadease** [comment]
Unauthorized exception is fixed now by downgrading to 0.9.3. Leaving this issue open as it should be clear why the 0.9.4. package is not imported properly.

**Comment by JorisHeadease** [comment]
Fixed with the release of v0.9.5.
